src/model.py

```diff
@@ -0,0 +1,96 @@
+import torch
+from torch import nn
+import pytorch_lightning as pl
+from transforms import Transforms
+from torchmetrics import Accuracy
+import torchvision.models as models
+
+
+
+class ResNetClassifier(pl.LightningModule):
+    def __init__(self, num_classes=2, learning_rate=1e-3, transform=None):
+        super(ResNetClassifier, self).__init__()
+        self.save_hyperparameters()
+
+        # Load data augmentation
+        self.transform = transform
+
+        # Load a pre-trained ResNet18 and modify the final layer
+        self.model = models.resnet18(pretrained=True)
+
+        # Freeze pre-trained layers
+        for param in self.model.parameters():
```

Comment:
I can  see that we are using the CNN as a feature extractor, [as in this tutorial](https://pytorch.org/tutorials/beginner/transfer_learning_tutorial.html#convnet-as-fixed-feature-extractor).

I propose to parametrize freezing of the convolutional layers with a `freeze` bool parameter. This is rather an extension, as your snippet is alright!